File: GoalCount/app/src/main/java/com/example/goalcount/CountDao.java
package com.example.goalcount;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface CountDao {
    @Insert
    void insert(CountEntry entry);

    @Delete
    void delete(CountEntry entry);

    @Query("SELECT * FROM count_entries ORDER BY timestamp DESC")
    LiveData<List<CountEntry>> getAllEntries();

    @Query("SELECT SUM(count) FROM count_entries")
    LiveData<Integer> getTotalCount();

    @Query("DELETE FROM count_entries")
    void deleteAll();
}

File: Count/app/src/main/java/com/example/count/data/CountDao.kt
package com.example.count.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface CountDao {
    @Query("SELECT * FROM counts ORDER BY timestamp DESC")
    fun getAllCounts(): Flow<List<CountEntry>>

    @Query("SELECT * FROM counts ORDER BY id DESC LIMIT 1")
    suspend fun getLastCount(): CountEntry?

    @Insert
    suspend fun insert(entry: CountEntry)

    @Query("DELETE FROM counts")
    suspend fun deleteAll()
}
